acq4/util/future.py

```diff
@@ -220,7 +228,11 @@ def waitFor(self, future: Future[WAITING_RETVAL_TYPE], timeout=20.0) -> Future[W
         """
         start = time.time()
         while True:
-            self.checkStop()
+            try:
+                self.checkStop()
+            except self.StopRequested:
+                future.stop(reason="parent task stop requested")
```

Comment:
this try/except wasn't actually necessary to get the correct behavior, but it seems like the right thing to do.